src/components/segmentedControl/index.tsx

```diff
@@ -0,0 +1,173 @@
+import _ from 'lodash';
+import React, {useRef, useState} from 'react';
+import {StyleSheet, StyleProp, ViewStyle, LayoutChangeEvent} from 'react-native';
+import Reanimated, {EasingNode, Easing as _Easing} from 'react-native-reanimated';
+import {Colors, BorderRadiuses, Spacings} from '../../style';
+import {asBaseComponent} from '../../commons/new';
+import View from '../view';
+import Segment, {SegmentItemProps} from './segment';
+
+const {interpolate: _interpolate, interpolateNode} = Reanimated;
+const interpolate = interpolateNode || _interpolate;
+const Easing = EasingNode || _Easing;
+const BORDER_WIDTH = 1;
+
+export type SegmentedControlProps = {
+  /**
+   * Array on segment labels.
+   */
+  labels?: SegmentItemProps[];
+  /**
+   * The color of the active segment.
+   */
+  activeColor?: string;
+  /**
+   * Callback for when index has change.
+   */
+  onChangeIndex?: (index: number) => void;
+  /**
+   * Initial index to be active.
+   */
+  initialIndex?: number;
+  /**
+   * The segmentedControl borderRadius
+   */
+  borderRadius?: number;
+  /**
+   * The background color of the segmentedControl
+   */
+  backgroundColor?: string;
+  /**
+   * The background color of the active segment
+   */
+  activeBackgroundColor?: string;
+  unActiveColor?: string;
+  outlineColor?: string;
+  outlineWidth?: number;
+  iconOnRight?: boolean;
+  style?: StyleProp<ViewStyle>;
+  testID?: string;
+};
+
+/**
+ * @description: SegmentedControl component for toggling two values or more
+ * @example: https://github.com/wix/react-native-ui-lib/blob/master/demo/src/screens/componentScreens/SegmentedControlScreen.tsx
+ */
+const SegmentedControl = (props: SegmentedControlProps) => {
+  const {
+    onChangeIndex,
+    initialIndex = 0,
+    style,
+    labels,
+    activeColor = Colors.primary,
+    borderRadius = BorderRadiuses.br100,
+    backgroundColor = Colors.grey80,
+    activeBackgroundColor = Colors.white,
+    unActiveColor = Colors.grey20,
+    outlineColor = Colors.grey60,
+    outlineWidth = BORDER_WIDTH
+  } = props;
+  const [selectedSegment, setSelectedSegment] = useState(-1);
+
+  const segments = useRef([] as {x: number; width: number}[]);
+  const segmentsCounter = useRef(0);
+  const animatedValue = new Reanimated.Value(selectedSegment);
```

Comment:
Same as the counter, the animated value is not stored in the state, so on every render it creates a new animated value which breaks your animation continuousness